File: Mobile/media_tracker_mobile/lib/models/lastfm/lastfm_user.dart
class LastFmUser {
  final String name;
  final int age;
  final bool isSubscriber;
  final String realName;
  final int playCount;
  final int artistCount;
  final int albumCount;
  final int trackCount;
  final String country;
  final String gender;
  final String profileUrl;
  final DateTime registeredAt;
  final String? imageUrl; // From the largest available image

  LastFmUser({
    required this.name,
    required this.age,
    required this.isSubscriber,
    required this.realName,
    required this.playCount,
    required this.artistCount,
    required this.albumCount,
    required this.trackCount,
    required this.country,
    required this.gender,
    required this.profileUrl,
    required this.registeredAt,
    this.imageUrl,
  });

  factory LastFmUser.fromJson(Map<String, dynamic> json) {
    final user = json['user'];
    final imageList = user['image'] as List<dynamic>;
    final largestImage = imageList.lastWhere(
      (img) => img['#text'] != null && img['#text'].isNotEmpty,
      orElse: () => {'#text': null},
    );

    return LastFmUser(
      name: user['name'],
      age: int.tryParse(user['age'] ?? '') ?? 0,
      isSubscriber: user['subscriber'] == '1',
      realName: user['realname'] ?? '',
      playCount: int.tryParse(user['playcount'] ?? '') ?? 0,
      artistCount: int.tryParse(user['artist_count'] ?? '') ?? 0,
      albumCount: int.tryParse(user['album_count'] ?? '') ?? 0,
      trackCount: int.tryParse(user['track_count'] ?? '') ?? 0,
      country: user['country'] ?? '',
      gender: user['gender'] ?? '',
      profileUrl: user['url'] ?? '',
      registeredAt: DateTime.fromMillisecondsSinceEpoch(
        int.parse(user['registered']['unixtime']) * 1000,
      ),
      imageUrl: largestImage['#text'],
    );
  }
}

File: Web/Media_tracker_Web/src/php/models/Last.FM/LastFmUser.php
<?php

class LastFmUser {
    public string $name;
    public int $age;
    public bool $isSubscriber;
    public string $realName;
    public int $playCount;
    public int $artistCount;
    public int $albumCount;
    public int $trackCount;
    public string $country;
    public string $gender;
    public string $profileUrl;
    public DateTime $registeredAt;
    public ?string $imageUrl;

    public function __construct(array $json) {
        $user = $json['user'];
        $imageList = $user['image'] ?? [];
        $image = end(array_filter($imageList, fn($img) => !empty($img['#text']))) ?: ['#text' => null];

        $this->name = $user['name'];
        $this->age = intval($user['age'] ?? 0);
        $this->isSubscriber = ($user['subscriber'] ?? '0') === '1';
        $this->realName = $user['realname'] ?? '';
        $this->playCount = intval($user['playcount'] ?? 0);
        $this->artistCount = intval($user['artist_count'] ?? 0);
        $this->albumCount = intval($user['album_count'] ?? 0);
        $this->trackCount = intval($user['track_count'] ?? 0);
        $this->country = $user['country'] ?? '';
        $this->gender = $user['gender'] ?? '';
        $this->profileUrl = $user['url'] ?? '';
        $this->registeredAt = new DateTime("@".intval($user['registered']['unixtime']));
        $this->imageUrl = $image['#text'];
    }
}
